napari/utils/misc.py

```diff
@@ -260,3 +260,53 @@ def all_subclasses(cls: Type) -> set:
     return set(cls.__subclasses__()).union(
         [s for c in cls.__subclasses__() for s in all_subclasses(c)]
     )
+
+
+_have_warned_psutil = False
+
+
+def resize_dask_cache(nbytes: int = None):
+    """Create or resize the dask cache for opportunistic caching.
+
+    The cache object is an instance of a :class`cachey.Cache`, and is made
+    available at :attr:`napari.utils.dask_cache`.
+
+    See `Dask oportunistic caching
+    <https://docs.dask.org/en/latest/caching.html>`_
+
+    Parameters
+    ----------
+    nbytes : int, optional
+        The desired size of the cache, in bytes, by default, napari wil attempt
+        to import ``psutil`` and use 80% of the available memory.  Otherwise it
+        will emit a warning and use 32GB.
+    """
+
+    from dask.cache import Cache
+    from napari import utils
+
+    if not nbytes:
+
+        try:
+            import psutil
+
+            # availalble RAM
+            nbytes = psutil.virtual_memory().available * 0.8
+        except ImportError:
+            import warnings
+
+            if not utils.misc._have_warned_psutil:
+                warnings.warn(
+                    'Could not import psutil to get available memory for '
+                    'caching. Run "pip install psutil" to automatically detect'
+                    ' memory. Or resize cache manually with '
+                    'napari.utils.resize_dask_cache'
+                )
+                utils.misc._have_warned_psutil = True
```

Comment:
as above — I don't think this should be needed.

Reply:
that would be great.  I did it cause I saw a double warning... but will go back and figure out what was going on there

Reply:
yeah, as mentioned above, I can't it to be a single warning no matter what I do (besides my explicit conditional), but since it's "supposed" to work like that (I even see the "once" filter in `warnings.filters`), maybe it's something in my environment.  I will just remove this code and let it potentially warn multiple times.